rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ScriptConsoleView.java

```diff
@@ -236,25 +245,37 @@ public void nodeUnselected() {
         this.label.show();
         this.nodeCanvas.hide();
         this.nodeUrl = null;
+        this.nodeSourceName = null;
+        this.nodeHostName = null;
     }
 
     public void nodeSelected(Node node) {
-        this.nodeLabel.setIcon(node.getNodeState().getIcon());
+        this.nodeLabel.setIcon(node.getIcon());
```

Comment:
The new lock status is no longer part of the `NodeState`. It is a dedicated field in the node abstraction. Since I need to have access to the NodeState but also the locking state in order to get the right image representation of the node, I have moved the `getIcon` method in the `Node` class. It explains why it is not longer required to use `getNodeState` in order to get the icon.